go/database/mpt/state_test.go

```diff
@@ -625,3 +625,44 @@ func TestState_parseCodes_ReadFailures(t *testing.T) {
 
 	}
 }
+
+func BenchmarkGoFileState_S5Archive_Flush_HighNumberOfDirtyNodes(b *testing.B) {
+	// running this test for all configurations at once takes too much time
+	// hence we use only the latest one
+	config := S5ArchiveConfig
+	cacheSize := 1_000_000
+	// number of accounts need to be ~1/10 smaller because we store leafs and branches
+	numAccounts := cacheSize - (cacheSize / 10)
+	dir := b.TempDir()
+	state, err := OpenGoFileState(dir, config, cacheSize)
+	if err != nil {
+		b.Fatalf("failed to open state, err %v", err)
+	}
+	addrs := getTestAddresses(numAccounts)
+	for i := 0; i < b.N; i++ {
+		for j, addr := range addrs {
+			if i%100_000 == 0 && i > 0 {
+				if _, _, err = state.UpdateHashes(); err != nil {
+					b.Fatalf("failed to update hashes: %v", err)
+				}
+			}
+			err = state.CreateAccount(addr)
+			if err != nil {
+				b.Fatalf("failed to create account: %v", err)
+			}
+			err = state.SetCode(addr, []byte{byte(j)})
+			if err != nil {
+				b.Fatalf("failed to set code: %v", err)
+			}
+		}
+
+		if err = state.Flush(); err != nil {
```

Comment:
should not we call only `Flush`() in the `b.N` for-loop?  At least for the bug with slow `Flush`(), I guess we are interested in performance of `Flush` only

Reply:
Indeed, those operations in addition to the flush are very costly, dominating the execution time. The problem got fixed in #842 .